src/cow-react/modules/swap/hooks/useFlowContext.ts

```diff
@@ -0,0 +1,275 @@
+  appData: AppDataInfo | null
+  wethContract: Weth | null
+  inputAmountWithSlippage: CurrencyAmount<Currency> | undefined
+  outputAmountWithSlippage: CurrencyAmount<Currency> | undefined
+  gnosisSafeInfo: SafeInfoResponse | undefined
+  recipient: string | null
+  recipientAddressOrName: string | null
+  deadline: number
+  ensRecipientAddress: string | null
+  allowsOffchainSigning: boolean
+  swapConfirmManager: SwapConfirmManager
+  isEthFlow: any
+  closeModals: () => void
+  addAppDataToUploadQueue: (update: AddAppDataToUploadQueueParams) => void
+  addOrderCallback: AddOrderCallback
+  dispatch: AppDispatch
+}
+
+export function useBaseFlowContext(): BaseFlowContext {
+  const { account, chainId, provider } = useWeb3React()
+  const { recipient } = useSwapState()
+  const { v2Trade: trade, allowedSlippage } = useDerivedSwapInfo()
+  const { allowsOffchainSigning, gnosisSafeInfo } = useWalletInfo()
+
+  const appData = useAppData({ chainId, allowedSlippage })
+  const closeModals = useCloseModals()
+  const addAppDataToUploadQueue = useUpdateAtom(addAppDataToUploadQueueAtom)
+  const addOrderCallback = useAddPendingOrder()
+  const dispatch = useDispatch<AppDispatch>()
+
+  const { address: ensRecipientAddress } = useENSAddress(recipient)
+  const recipientAddressOrName = recipient || ensRecipientAddress
+  const [deadline] = useUserTransactionTTL()
+  const wethContract = useWETHContract()
+  const swapConfirmManager = useSwapConfirmManager()
+  const isEthFlow = useIsEthFlow()
+
+  const { INPUT: inputAmountWithSlippage, OUTPUT: outputAmountWithSlippage } = computeSlippageAdjustedAmounts(
+    trade,
+    allowedSlippage
+  )
+
+  return {
+    chainId,
+    account,
+    provider,
+    trade,
+    appData,
+    wethContract,
+    inputAmountWithSlippage,
+    outputAmountWithSlippage,
+    gnosisSafeInfo,
+    recipient,
+    recipientAddressOrName,
+    deadline,
+    ensRecipientAddress,
+    allowsOffchainSigning,
+    swapConfirmManager,
+    isEthFlow,
+    closeModals,
+    addAppDataToUploadQueue,
+    addOrderCallback,
+    dispatch,
+  }
+}
+
+type BaseGetFlowContextProps = {
+  baseProps: BaseFlowContext
+  conditionalCheck: boolean
+  sellToken: Currency
+  kind: OrderKind
+}
+
+interface GetCommonFlowContextProps extends BaseGetFlowContextProps {
+  contract: any
+}
+
+interface GetSwapFlowContextProps extends BaseGetFlowContextProps {
+  contract: GPv2Settlement | null
+}
+
+interface GetEthFlowContextProps extends BaseGetFlowContextProps {
+  contract: CoWSwapEthFlow | null
+}
+
+export function getFlowContext({
+  baseProps,
+  conditionalCheck,
+  sellToken,
+  kind,
+  contract,
+}: GetSwapFlowContextProps): SwapFlowContext
+export function getFlowContext({
+  baseProps,
+  conditionalCheck,
+  sellToken,
+  kind,
+  contract,
+}: GetEthFlowContextProps): EthFlowContext
+export function getFlowContext({
+  baseProps,
+  conditionalCheck,
+  sellToken,
+  kind,
+  contract,
+}: GetCommonFlowContextProps): any {
```

Comment:
Why `any`? Aren't the types returned known? (null | SwapFlowContext | EthFlowContext)

Reply:
this doesn't work with overload functions as there isn't any real inheritance in those 2 contract types and function overloading needs `any`:
https://www.tutorialsteacher.com/typescript/function-overloading

Reply:
it's a function overload thing, [see https://www.tutorialsteacher.com/typescript/function-overloading](https://www.typescriptlang.org/docs/handbook/2/functions.html#writing-good-overloads)

truthfully i'd like to avoid them but the return type is different contract: EthFlowContract | GpV2Contract so AFAIK it needs to be overloaded in such a way